Automattic/wp-calypso
# Add product slug to query param for products endpoint

Related to # https://github.com/Automattic/martech/issues/2722#issue-2137938101

## Proposed Changes

Add product_slug as a query parameter to the products endpoint call during the checkout process. 

## Testing Instructions

1. Checkout this branch 
2. Open Chrome
3. Open Developer tools and navigate to the Network Tab
4. Filter requests by product
5. Login to Calypso
6. Navigate to plans and select a plan that can be purchased (Starter, Creator etc.)
7. Click the back button in the top left to navigate back to the dashboard and leave the items in the cart
8. Go to add-ons and add the 50GB storage to the cart 
9. In the Network tab you should see a request being to the products endpoint, however, it should now include the product_slugs query parameter with the slugs for the selected products.
10. Only the requested products should be included in the endpoint response
11. The response time should be in the milliseconds.

![image](https://github.com/Automattic/wp-calypso/assets/25906001/c34424f6-3b30-412b-a03a-ca8341c28759)
